lms/djangoapps/discussion/templates/discussion/discussion_profile_page.html

```diff
@@ -5,74 +5,90 @@
-            $el: $('.discussion-user-threads')
+            $el: $('.discussion-user-profile-board')
         },
-        ${profile_page_context | n, dump_js_escaped_json}
+        profile_page_context
     ));
 </%static:require_module>
 </%block>
 
 <%include file="../courseware/course_navigation.html" args="active_page='discussion'" />
 
-<section class="container">
+<%block name="content">
+<section class="discussion inline-discussion discussion-user-profile-board container">
     <header class="page-header">
         <div class="page-header-main">
             <div class="sr-is-focusable" tabindex="-1"></div>
-            <h2 class="hd hd-2 page-title">${_("Discussion")}</h2>
+            <div>
+                <%def name="span(num)"><span class="discussion-count">${num}</span></%def>
+                <span class="user-name"><a href="${learner_profile_page_url}">${django_user.username}</a></span>
+                <span class="discussion-profile-info">
+                    <span class="user-roles">${", ".join(_(role_name) for role_name in django_user_roles)}</span>
+                </span>
+                <div class="discussion-profile-count">
+                    <span class="discussion-profile-info threads-count">${ungettext('%s discussion started', '%s discussions started', profiled_user['threads_count']) % span(profiled_user['threads_count'])}</span>
+                    <span class="discussion-profile-info  comments-count">${ungettext('%s comment', '%s comments', profiled_user['comments_count']) % span(profiled_user['comments_count'])}</span>
+                </div>
+            </div>
         </div>
     </header>
     <div class="page-content">
-        <div class="layout layout-1t2t">
-            <aside class="forum-nav layout-col layout-col-a" role="complementary" aria-label="${_("Discussion thread list")}">
-                <nav class="user-profile" aria-label="${_('User Profile')}">
-
-                    <article class="sidebar-module discussion-sidebar">
-                        <%include file="_user_profile.html" />
-                    </article>
-
-                </nav>
-            </aside>
-
-            <main id="main" aria-label="Content" tabindex="-1" class="discussion-column layout-col layout-col-b">
-                <div class="course-content discussion-user-threads" data-course-id="${course.id}"
-                     data-course-name="${course.display_name_with_default}"
-                     data-threads="${threads}" data-user-info="${user_info}"
-                     data-page="${page}" data-num-pages="${num_pages}">
-                </div>
-            </main>
-        </div>
+        <main id="main" aria-label="Content" tabindex="-1" class="discussion-column">
+            <div class="course-content discussion-module discussion-user-threads"
+                data-course-id="${course.id}"
+                data-course-name="${course.display_name_with_default}"
+                data-threads="${threads}"
```

Comment:
Made this comment on an earlier review, but it was lost: why do pass some of this data to `DiscussionPageProfileFactory` as context directly as a JS object (e.g., course ID, threads, course name, num pages... most of the same data as is listed here, actually) but then set it as data attrs on this node as well?

This looks really messy when inspecting this page (screenshot below) and to me it violates a design principle of web applications - the DOM is not a data layer, and shouldn't be used for storage/messaging, only presentation. AFAIK that's one of the reasons we adopted the factory pattern, right @andy-armstrong?

But maybe this has to be the approach, for architectural/historical reasons... I might be way off here. Let me know your thoughts.

![](https://i.bjacobel.com/20161202-pn332.png)